docs/reference/pip_install.rst

```diff
@@ -36,6 +39,33 @@ which depends on foo::
     ...
     Installing collected packages foo, baz, bar
 
+
+Prior to v6.1.0, pip made no commitments about install order.
+
+The decision to install topologically is based on the principle that
+installations should proceed in a way that leaves the environment usable at each
+step. This has two main practical benefits:
+
+1. Concurrent use of the environment during the install is more likely to work.
+2. A failed install won't leave broken packages.  Although pip would like to
```

Comment:
@pfmoore I'm not sure, hence I'm inclined to just add the "less likely" language for now